### PR TITLE
Migration to typescript

### PR DESCRIPTION
**Pull request by Walkeryr**
## Overview
Added support for compilation project code with TypeScript.

## Problem
We want to introduce TypeScript compilation for better developer experience.

## Change log
* Configured webpack to compile TypeScript
* Updated linting to handle TypeScript code
* Updated jest configuration to run TypeScript code
* Added alias support for import paths
* Updated all dependencies

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). This requires a major version bump.
- [ ] This change requires a documentation update.

## How can this be tested?
Run it locally with `make run` or deploy it to your sandbox with `make deploy`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by ferencdev — APPROVED** [review]
:ok_hand: 

Should we rename the repo to typescript after merging?

**Review by innocentugwunna — APPROVED** [review]
Typescript 💌

**Comment by Walkeryr** [comment]
@ferencdev yeah, I had the same thought. What do you think @me2resh?

**Comment by EikeDawid** [comment]
@Walkeryr @ferencdev the current naming is based on the runtime (nodejs) and not the language (javascript) so i think the name can stay the same?

**Comment by ferencdev** [comment]
> @Walkeryr @ferencdev the current naming is based on the runtime (nodejs) and not the language (javascript) so i think the name can stay the same?

Just wondered because we have this:
https://github.com/healthbridgeltd/typescript-serverless-bootstrap

**Comment by Walkeryr** [comment]
Keeping old name then and added more info in the readme file.

File: package.json [hunk 1]
```diff
@@ -4,13 +4,19 @@
...
   "dependencies": {
-    "aws-sdk": "^2.677.0",
-    "source-map-support": "^0.5.19"
+    "aws-sdk": "2.787.0",
+    "source-map-support": "0.5.19"
   },
   "scripts": {
-    "lint": "standard --env jest | standard-reporter --stylish",
-    "lint:fix": "standard --env jest --fix",
+    "lint": "standardx ./src/**/*.ts --env jest | standard-reporter --stylish",
```
Comment (EikeDawid) [inline package.json:18]: the BFF's are using `eslint` - with `standard` and also some custom rules:
https://github.com/healthbridgeltd/patient-consultation-experience/blob/master/.eslintrc.js

Does it make sense to align these?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  "include": [
+    "./**/*.ts"
+  ],
+  "exclude": [
+    "node_modules/**/*",
+    ".webpack/**/*",
+    ".vscode/**/*"
+  ]
+}
```
Comment (EikeDawid) [inline tsconfig.json:27]: similar question as for linter: the BFF tsconfig is different (not sure which is better though) - is that another thing we should align?

https://github.com/healthbridgeltd/patient-consultation-experience/blob/master/tsconfig.json

**Review by EikeDawid — APPROVED** [review]
added a couple of comments in regards to alignment across projects - however i don't think that they should block this PR. However would be good to get them standardised in general

File: package.json (hunk 1 above)
Comment (Walkeryr) [reply package.json:18]: I'll check it, thanks!

File: src/cmd/lambdas/hello-world/app.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
-const messages = require('../../../internal/app/hello-world/messages')
+import messages from "@/internal/app/hello-world/messages"
+
 let response
 
 /**
  * Hello World
  */
-exports.lambdaHandler = async (event, context) => {
+export const lambdaHandler = async () => {
```
Comment (mgarciadelojo) [inline src/cmd/lambdas/hello-world/app.ts:8]: Would it make sense to use a middleware system as we use in the prescription tribe? We are planning on adding it to our bootstrap project too i.e.

https://github.com/healthbridgeltd/payment-service/blob/master/src/middlewares/index.ts

This example uses already the zava api response standard by default.

Comment (Walkeryr) [reply src/cmd/lambdas/hello-world/app.ts:8]: @mgarciadelojo yep, I think that would be a good addition for the bootstrap, though in another PR 🙂. @healthbridgeltd/questionnaires what do you think about adding middleware system?

File: package.json (hunk 1 above)
Comment (Walkeryr) [reply package.json]: ✅

File: tsconfig.json (hunk 2 above)
Comment (Walkeryr) [reply tsconfig.json:26]: Ok, removed `allowJs` and `checkJs` compiler options and made checks stricter. Configs not fully identical, but became a bit closer.

**Pull request merged**